### DISCUSSION TITLE
Any love for nuget?

### DISCUSSION THREAD

**Discussion by madfu** [applejag/Newtonsoft.Json-for-Unity#126] [Ideas]
hi.

i know from using this excellent repo that the focus is on helping the Unity community.
i've especially appreciated the work on the UPM.  that was really fun to get working.

after checking the wiki and seeing only one mention of nuget, i feel the need to ask, are there any plans for releasing a nuget package?

i am particularly interested in this, as i usually put most of my networking and security code for my Unity projects in DLLs and nuget is often the easiest way to stay up-to-date on a library.  right now, i'm in the midst of integrating Newtonsoft.Json-for-Unity into a fork of Nethereum (which is another collection of DLLs), and from my research, i haven't found your version of Newtonsoft anywhere within nuget.

many thanks to all the contributors for their excellent work on this project, and thank you for your time.

-madfu

**Comment by applejag** [comment]
Ah cool you are the first one to take use of the discussions :) And big thanks for checking the wiki before asking :)

The procedure I recommend:

1. Use the [official Newtonsoft.Json NuGet package](https://www.nuget.org/packages/Newtonsoft.Json/) in your libraries that you build outside of Unity
2. When importing your own library's DLLs, skip adding the `Newtonsoft.Json.dll` from the NuGet
3. Import my Newtonsoft.Json-for-Unity UPM package

This works because the assembly name is the same (`Newtosoft.Json`) and I've used the same signing key ([Src/IdentityPublicKey.snk](https://github.com/jilleJr/Newtonsoft.Json-for-Unity/blob/master/Src/IdentityPublicKey.snk)), albeit only a public key but Unity does not verify the key and only uses it to match the [assembly strong name](https://docs.microsoft.com/en-us/dotnet/standard/assembly/strong-named) when compiling.

While I could post the project as a NuGet package, I refrain due to the fact that I wish users to include my package via UPM and not by adding the DLL binaries directly into their repos.

Are you OK with that explanation? There's kind of minimal work to create a NuGet package, but I do not see much gain for it.
